test/time_sliding_window_test.rb

```diff
@@ -0,0 +1,74 @@
+require 'test_helper'
+
+class TestTimeSlidingWindow < Minitest::Test
+  def setup
+    @sliding_window = ::Semian::ThreadSafe::TimeSlidingWindow.new(0.5, -> { Time.now.to_f * 1000 }) # Timecop doesn't work with a monotonic clock
+    @sliding_window.clear
+  end
```

Comment:
I would throw a `Timecop.freeze` in the setup as well as a `Timecop.return` in the teardown to prevent any test flakyness caused by shared machines executing the tests.